src/libpgmoneta/wal.c

```diff
@@ -420,6 +427,11 @@ pgmoneta_wal(int srv, char** argv)
       pgmoneta_consume_data_row_messages(ssl, socket, buffer, &end_of_timeline_response);
       timeline = atoi(pgmoneta_query_response_get_data(end_of_timeline_response, 0));
       xlogpos = pgmoneta_query_response_get_data(end_of_timeline_response, 1);
+
+      // assign xlogpos at the end of the steaming to LSN
+      memset(config->servers[srv].current_wal_lsn, 0, MISC_LENGTH);
+      snprintf(config->servers[srv].current_wal_lsn, MISC_LENGTH, "%s", xlogpos);
```

Comment:
Same here - new function

Reply:
I don't think we even need these here on second thought. We are getting this from the server, which is not necessarily where we start streaming when the next timeline starts - in-segment offset should be removed first - so I think updating the lsn when sending START_REPLICATION in the next timeline is enough.

Reply:
We need to update the LSN along the current timeline as well

Reply:
It is updated.. This xlogpos is just server's cue on where we should start on the next timeline.

Reply:
The LSN is updated by the new function every time we receives a message data. So it is updated along the timeline, I believe.

Reply:
I’ll remove this block of code.